gcp/src/main/java/org/apache/iceberg/gcp/auth/GoogleAuthManager.java

```diff
@@ -0,0 +1,160 @@
+          this.credentials = GoogleCredentials.fromStream(credentialsStream).createScoped(scopes);
+        }
+      } else {
+        LOG.info("Using Application Default Credentials with scopes: {}", scopesString);
+        this.credentials = GoogleCredentials.getApplicationDefault().createScoped(scopes);
+      }
+    } catch (IOException e) {
+      throw new UncheckedIOException("Failed to load Google credentials", e);
+    }
+    this.initialized = true;
+  }
+
+  /**
+   * Initializes and returns a short-lived session, typically for fetching configuration. This
+   * implementation reuses the long-lived catalog session logic.
+   */
+  @Override
+  public AuthSession initSession(RESTClient initClient, Map<String, String> properties) {
+    return catalogSession(initClient, properties);
+  }
+
+  /**
+   * Returns a long-lived session tied to the catalog's lifecycle. This session uses Google
+   * Application Default Credentials or a specified service account.
+   *
+   * @param sharedClient The long-lived RESTClient (not used by this implementation for credential
+   *     fetching).
+   * @param properties Configuration properties for the auth manager.
+   * @return A {@link GoogleAuthSession}.
+   * @throws UncheckedIOException if credential loading fails.
+   */
+  @Override
+  public AuthSession catalogSession(RESTClient sharedClient, Map<String, String> properties) {
+    initialize(properties);
+    Preconditions.checkState(
+        credentials != null, "GoogleAuthManager not initialized or failed to load credentials");
+    return new GoogleAuthSession(credentials);
+  }
+
+  /** Returns a session for a specific context. Defaults to the catalog session. */
+  @Override
+  public AuthSession contextualSession(SessionCatalog.SessionContext context, AuthSession parent) {
+    // For GCP, tokens are typically not context-specific in this manner.
+    // Re-using the parent (which should be a GoogleAuthSession) is appropriate.
+    // Or, if properties for a specific context were available, a new GoogleAuthSession could be
+    // derived.
+    if (parent instanceof GoogleAuthSession) {
+      return parent;
+    }
+    // Fallback to a new catalog-level session if the parent is not a GoogleAuthSession for some
+    // reason.
+    // This would require properties to be available or a default initialization.
+    LOG.warn(
+        "Parent session is not a GoogleAuthSession. Creating a new default catalog session. This might not be intended.");
+    return catalogSession(
+        null, Collections.emptyMap()); // Assuming default ADC without specific props
+  }
+
+  /** Returns a session for a specific table or view. Defaults to the catalog session. */
+  @Override
+  public AuthSession tableSession(
+      TableIdentifier table, Map<String, String> properties, AuthSession parent) {
+    // Similar to contextualSession, GCP tokens are generally not table-specific.
+    if (parent instanceof GoogleAuthSession) {
```

Comment:
1. Same as above, this will always be true.
2. In this method, `properties` are sent by the catalog server. E.g. some servers could return a more specific scope here that is required to access tables. Are we OK ignoring these properties?

Reply:
What should I do if they send reduced scope ? Recreate GoogleAuthSession ?

Reply:
Yes it could make sense to create a downscoped GoogleAuthSession. But it would also probably make sense to cache the downscoped session as well, to avoid creating too many sessions.